util/predicates/secret_predicates.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package predicates
+
+import (
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	addonsv1 "sigs.k8s.io/cluster-api/exp/addons/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// AddonsSecretCreateUpdate returns a predicate that returns true for a Secret create event if in addons Secret type
+func AddonsSecretCreateUpdate(logger logr.Logger) predicate.Funcs {
```

Comment:
Do you expect there to be other consumers of this predicate? If not, it might make sense to keep it internal to the exp/addons/controllers package.

Reply:
It agree we don't, considering we are doing things specific to Secret type CRS is using. So moved it to exp.